packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -390,13 +553,25 @@ fn schema_to_stream_bytes(schema: &Schema) -> Vec<u8> {
     stream_bytes
 }
 
-fn array_to_user_errors(array: mv8::Value<'_>) -> Vec<UserError> {
-    let fallback = format!("Unparsed: {:?}", array);
-
-    if let mv8::Value::Array(array) = array {
-        let errors = array
-            .elements()
-            .map(|e: mv8::Result<'_, mv8::Value<'_>>| e.map(|e| UserError(format!("{e:?}"))))
+fn array_to_user_errors<'s>(
+    scope: &mut v8::HandleScope<'s>,
+    array: v8::Local<'s, v8::Value>,
+) -> Vec<UserError> {
+    let fallback = format!("Unparsed: {array:?}");
+
+    if array.is_array() {
+        let array: v8::Local<'s, v8::Array> = array
+            .try_into()
+            .expect("array conversion to never fail as we just checked it was the right type");
```

Comment:
This isn't how expect statements are propagated. They're not exposed like "Expected array conversion to...", so can we phrase it to "Array conversion failed..." but if we are sure it can't